src/main/java/com/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -52,6 +57,15 @@ public BookDto update(Long id, CreateBookRequestDto requestDto) {
         Book book = bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("There is no book with the id " + id));
         bookMapper.updateBook(requestDto, book);
+        book.getCategories().clear();
+        setCategories(requestDto.categoriesIds(), book);
         return bookMapper.toDto(bookRepository.save(book));
     }
+
+    private void setCategories(Set<Long> categoriesIds, Book book) {
+        for (Long id : categoriesIds) {
+            Category category = categoryRepository.getReferenceById(id);
+            book.getCategories().add(category);
+        }

```

Comment:
you can use stream and return Set<Category> and avoid `book.getCategories().clear();`